### PR TITLE
creatures.py

### PR DESCRIPTION
**Pull request by 2dete**
Created creatures.py, started to create some basic classes for some basic elements of game

### CODE REVIEW

File: creatures.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+class Weapon:
+    def __init__(self, x, y):
+        self.hp = random.randint(1, 10)
+        self.rect = pygame.Rect((x, y, 20, 20))  # we need this for collision
+
+
+class Shield:
+    def __init__(self, x, y):
+        self.hp = random.randint(1, 10)
+        self.rect = pygame.Rect((x, y, 20, 20))  # we need this for collision
```
Comment (LinSharp) [inline creatures.py:29]: Do we need collition for shield? We can just assume player shielding when button is pressed.

File: creatures.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        self.hp = random.randint(10, 20)
+        self.attack = random.randint(1, 5)
+        self.xp = 10
+        self.rect = pygame.Rect((x, y, 20, 20))    # we need this for collision
+
+
+class Hero(Creature):
+    def __init__(self):
+        super().__init__()
+        self.hp = random.randint(50, 100)
```
Comment (LinSharp) [inline creatures.py:16]: HP usually isn't random in games. So a static number for now like 100 will do.

File: creatures.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import random
+import pygame
+
+
+class Creature:
+    def __init__(self, x, y):
+        self.hp = random.randint(10, 20)
```
Comment (LinSharp) [inline creatures.py:7]: HP usually isn't random in games. So a static number for now like 100 will do.

File: creatures.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        self.attack = random.randint(1, 5)
+        self.xp = 10
```
Comment (LinSharp) [inline creatures.py:9]: XP is only for player. Maybe change to Level (lvl).

File: creatures.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        self.rect = pygame.Rect((x, y, 20, 20))    # we need this for collision
+
+
+class Hero(Creature):
+    def __init__(self):
+        super().__init__()
+        self.hp = random.randint(50, 100)
+        self.speed = random.randint(2, 5)
```
Comment (LinSharp) [inline creatures.py:17]: Speed isn't random either. :)

File: creatures.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import pygame
+
+
+class Creature:
+    def __init__(self, x, y):
+        self.hp = random.randint(10, 20)
+        self.attack = random.randint(1, 5)
+        self.xp = 10
+        self.rect = pygame.Rect((x, y, 20, 20))    # we need this for collision
+
```
Comment (LinSharp) [inline creatures.py:11]: No speed for creature :(

**Comment by LinSharp** [comment]
All creatures would need to be drawn on screen, so self.sprite should be added

File: creatures.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+class Weapon:
+    def __init__(self, x, y):
+        self.hp = random.randint(1, 10)
+        self.rect = pygame.Rect((x, y, 20, 20))  # we need this for collision
+
+
+class Shield:
+    def __init__(self, x, y):
+        self.hp = random.randint(1, 10)
```
Comment (LinSharp) [inline creatures.py:28]: maybe change to `self.def` As shields protect.

**Comment by LinSharp** [comment]
Besides creatures you have items too. Maybe rename file to game_objects.py?

File: creatures.py (hunk 1 above)
Comment (2dete) [reply creatures.py:29]: I think that at the end we don't need shield at this moment.

File: creatures.py (hunk 2 above)
Comment (2dete) [reply creatures.py:16]: agree

File: creatures.py (hunk 6 above)
Comment (2dete) [reply creatures.py:11]: Here the Creature class can be used even for static elements i.e. trash bin or rock

File: creatures.py (hunk 7 above)
Comment (2dete) [reply creatures.py:28]: remove shield class all at onece, for now

**Review by LinSharp — APPROVED** [review]
Hero has no damage parameter.

**Pull request merged**